api/src/test/java/org/openmrs/module/locationbasedaccess/aop/PatientSearchAdviserTest.java

```diff
@@ -0,0 +1,207 @@
+package org.openmrs.module.locationbasedaccess.aop;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.Location;
+import org.openmrs.Patient;
+import org.openmrs.User;
+import org.openmrs.api.LocationService;
+import org.openmrs.api.PatientService;
+import org.openmrs.api.UserService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.locationbasedaccess.aop.common.AOPContextSensitiveTest;
+import org.openmrs.module.locationbasedaccess.aop.common.TestWithAOP;
+import org.openmrs.module.locationbasedaccess.aop.interceptor.PatientServiceInterceptorAdvice;
+import java.util.List;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertEquals;
+
+public class PatientSearchAdviserTest extends AOPContextSensitiveTest {
+
+
+    private UserService userService;
+    private PatientService patientService;
+    private LocationService locationService;
+    private static final String XML_FILENAME_WITH_PERSON_DATA = "include/PatientTestData.xml";
+    private static final String XML_FILENAME_WITH_ADMIN_USER_DATA = "include/AdminUserData.xml";
+
+    @Override
+    protected void setInterceptorAndServices(TestWithAOP testCase) {
+        testCase.setInterceptor(PatientServiceInterceptorAdvice.class);
+        testCase.addService(PatientService.class);
+    }
+
+    @Before
+    public void setUp()  throws Exception {
+        userService = Context.getUserService();
+        patientService = Context.getPatientService();
+        locationService = Context.getLocationService();
+        executeDataSet(XML_FILENAME_WITH_ADMIN_USER_DATA);
+        executeDataSet(XML_FILENAME_WITH_PERSON_DATA);
+    }
+
+    @Test
+    public void getPatients_adminShouldGetPatientsFromAllLocations() {
+        Context.authenticate("admin", "test");
+        User authenticatedUser = Context.getAuthenticatedUser();
+        assertNotNull(authenticatedUser);
+        assertTrue(authenticatedUser.isSuperUser());
+
+        Location location1 = locationService.getLocation(1);
+        Location location2 = locationService.getLocation(2);
+
+        List<Patient> patientList = patientService.getPatients("Patient");
+        assertEquals(patientList.size(), 3);
+        assertEquals(patientList.get(0).getAttribute(99).getValue(), location1.getUuid());
```

Comment:
What is the 99 magic number?

Reply:
Oh, Got it from testing dataset :smile:  
Just changed to 10 which can differentiate from others easily.